.github/workflows/main.yml

```diff
@@ -39,3 +39,20 @@ jobs:
       - uses: hadolint/hadolint-action@v1.5.0
         with:
           failure-threshold: error
+
+  docker-publish:
+    if: github.ref == 'refs/heads/main'
```

Comment:
I think this will work fine, but the standard approach would be to make a new workflow file and specify that this only runs on the `main` branch c.f.
https://github.com/opensafely-core/python-docker/blob/main/.github/workflows/build_and_publish.yaml

(Edit: unless this is deliberate because the `needs` dependencies don't work across files, which is quite possible)

Reply:
Yes -- I wanted dependencies on the other stuff.